### PR TITLE
Dashboard/stats none

### PR DESCRIPTION
**Pull request by Wardormeur**
Add info message to inform the user is not having data to display
Relevant commit is https://github.com/CoderDojo/cp-zen-frontend/commit/e272d6c58a8da68bd49314f46a971441130bee8b

### CODE REVIEW

File: src/dashboard/cd-dashboard-stats.vue [hunk 1]
```diff
@@ -1,30 +1,37 @@
...
+              </div>
             </div>
           </div>
         </div>
-        <h4>Gender repartition</h4>
+        <div v-else class="cd-dashboard-stats__chart-pie--filler cd-filler cd-filler--grey-bg"/>
+      </div>
+      <div v-else>
+        <p>{{ $t('No statistics are available at the moment.') }}
+        {{ $t('The more Zen will be used, the more you\'ll find about your Dojo!') }}</p>
```
Comment (heyitsmaimai) [inline src/dashboard/cd-dashboard-stats.vue]: should be "The more Zen is used..."

File: src/dashboard/cd-dashboard-stats.vue [hunk 2]
```diff
@@ -87,9 +91,9 @@
         )).body.length;
       },
       async getDojoUsers() {
-        this.dojoUsers = (await DojoService.getDojoUsers(this.dojoId, { userType: 'attendee-u13', fields: ['gender'] })).body.response;
-        this.dojoUsers = this.dojoUsers.concat((await DojoService.getDojoUsers(this.dojoId, { userType: 'attendee-o13', fields: ['gender'] })).body.response);
-        this.genders = this.dojoUsers.reduce((red, user) => {
+        let dojoUsers = (await DojoService.getDojoUsers(this.dojoId, { userType: 'attendee-u13', fields: ['gender'] })).body.response;
```
Comment (heyitsmaimai) [inline src/dashboard/cd-dashboard-stats.vue]: any reason for using let instead of const?

File: src/dashboard/cd-dashboard-stats.vue (hunk 1 above)
Comment (Wardormeur) [reply src/dashboard/cd-dashboard-stats.vue]: Engrish plz.
Yes, very likely

File: src/dashboard/cd-dashboard-stats.vue (hunk 2 above)
Comment (Wardormeur) [reply src/dashboard/cd-dashboard-stats.vue]: It is reassigned the line after

Comment (heyitsmaimai) [reply src/dashboard/cd-dashboard-stats.vue]: ah ok

File: src/dashboard/cd-dashboard-stats.vue [hunk 3]
```diff
@@ -1,30 +1,37 @@
...
-            </g>
-          </svg>
-          <div class="cd-dashboard-stats__circle-legends">
-            <div v-for="gender in genderStats" class="cd-dashboard-stats__circle-legend">
-              <div class="cd-dashboard-stats__circle-legend-color" :class="['cd-dashboard-stats__circle--' + gender.name]"></div>
-              <div stroke="#51c5cf" stroke-width="2px" text-anchor="middle" alignment-baseline="middle"> {{ gender.name }}</div>
+      <h3 class="cd-dashboard-stats__header">{{ $t('Dojo Stats') }}</h3>
+      <h4 class="cd-dashboard-stats__category">Youth</h4>
+      <div class="cd-dashboard-stats__charts" 
+        v-if="(bookedChildren === null || bookedChildren.length !== 0) && ((dojoUsers === null || dojoUsers.length !== 0) && (genders === null || genders.length !== 0))">
```
Comment (DanielBrierton) [inline src/dashboard/cd-dashboard-stats.vue]: Computed property perhaps? This is very long 😅

File: src/dashboard/cd-dashboard-stats.vue [hunk 4]
```diff
@@ -1,30 +1,37 @@
...
+        <div class="cd-dashboard-stats__chart-number" v-if="bookedChildren !== null">
+          <span class="cd-dashboard-stats__description" v-html="$t('{nbKids} kids attended your events', { nbKids: numberStatText })"></span>
```
Comment (DanielBrierton) [inline src/dashboard/cd-dashboard-stats.vue]: `{nbKids}` wouldn't be very obvious to a translator. Just `{num}` or `{numKids}` would be better.

File: src/dashboard/cd-dashboard-stats.vue [hunk 5]
```diff
@@ -1,30 +1,37 @@
...
+        </div>
+        <div v-else class="cd-dashboard-stats__chart-number--filler cd-filler cd-filler--grey-bg"/>
+        <div class="cd-dashboard-stats__chart-pie" v-if="(dojoUsers !== null && dojoUsers.length > 0) || (genders !== null && genders.length > 0)">
```
Comment (DanielBrierton) [inline src/dashboard/cd-dashboard-stats.vue]: Moar long v-if conditionals

File: src/dashboard/cd-dashboard-stats.vue [hunk 6]
```diff
@@ -1,30 +1,37 @@
...
+            <svg viewBox="0 0 32 32">
+              <g v-for="gender in genderStats">
+                <circle r="16" cx="16" cy="16" :style="{ strokeDashoffset: -gender.prevValue, strokeDasharray: `${gender.perc} 100`}"
+                  :class="['cd-dashboard-stats__circle--' + gender.name]">
+                </circle>
+              </g>
+            </svg>
+            <div class="cd-dashboard-stats__circle-legends">
+              <div v-for="gender in genderStats" class="cd-dashboard-stats__circle-legend">
+                <div class="cd-dashboard-stats__circle-legend-color" :class="['cd-dashboard-stats__circle--' + gender.name]"></div>
```
Comment (DanielBrierton) [inline src/dashboard/cd-dashboard-stats.vue]: String concat again. I am disappoint

**Pull request merged**